test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+    const BRLCMock: ContractFactory = await ethers.getContractFactory("ERC20UpgradeableMock");
+    brlcMock = await upgrades.deployProxy(BRLCMock, ["BRL Coin", "BRLC", 6]);
+    await brlcMock.deployed();
+
+    // Deploy the being tested contract
+    const CardPaymentProcessor: ContractFactory = await ethers.getContractFactory("CardPaymentProcessorUpgradeable");
+    cardPaymentProcessor = await upgrades.deployProxy(CardPaymentProcessor, [brlcMock.address]);
+    await cardPaymentProcessor.deployed();
+
+    // Get user accounts
+    [deployer, user1, user2] = await ethers.getSigners();
+  });
+
+  it("The initialize function can't be called more than once", async () => {
+    await expect(cardPaymentProcessor.initialize(brlcMock.address))
+      .to.be.revertedWith(REVERT_MESSAGE_IF_CONTRACT_IS_ALREADY_INITIALIZED);
+  });
+
+  describe("Function 'setRevocationCounterMaximum()'", async () => {
+    const revocationCounterNewValue: number = 123;
+    const revocationCounterMaximumDefaultValue: number = 255;
+
+    it("Is reverted if is called not by the owner", async () => {
+      await expect(cardPaymentProcessor.connect(user1).setRevocationCounterMaximum(
+        revocationCounterNewValue
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_CALLER_IS_NOT_OWNER);
+    });
+
+    it("Is reverted if the new value is zero", async () => {
+      await expect(cardPaymentProcessor.setRevocationCounterMaximum(0))
+        .to.be.revertedWith(REVERT_MESSAGE_IF_CALLER_IS_NEW_REVOCATION_COUNTER_MAXIMUM_VALUE_IS_ZERO);
+    });
+
+    it("Emits the correct event, changes the revocation counter maximum properly", async () => {
+      expect(await cardPaymentProcessor.revocationCounterMaximum()).to.equal(revocationCounterMaximumDefaultValue);
+      await expect(cardPaymentProcessor.setRevocationCounterMaximum(
+        revocationCounterNewValue
+      )).to.emit(
+        cardPaymentProcessor,
+        "SetRevocationCounterMaximum"
+      ).withArgs(
+        revocationCounterMaximumDefaultValue,
+        revocationCounterNewValue
+      );
+    });
+
+    it("Does not emit events if the new value equals the old one", async () => {
+      await expect(cardPaymentProcessor.setRevocationCounterMaximum(
+        revocationCounterMaximumDefaultValue
+      )).not.to.emit(
+        cardPaymentProcessor,
+        "SetRevocationCounterMaximum"
+      );
+    });
+  });
+
+  describe("Function 'makePayment()'", async () => {
+    let payment: TestPayment;
+    let authorizationId: string;
+    let correlationId: string;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 123,
+        account: user1,
+        amount: 234,
+        status: PaymentStatus.Nonexistent,
+        makingPaymentCorrelationId: 345,
+      };
+      authorizationId = createBytesString(payment.authorizationId, BYTES16_LENGTH);
+      correlationId = createBytesString(payment.makingPaymentCorrelationId, BYTES16_LENGTH);
+      await setUpContractsForPayments([payment])
```

Comment:
Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
      await setUpContractsForPayments([payment]);
```